test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -593,7 +594,8 @@ private NodeAndClient buildNode(int nodeId, long seed, Settings settings,
             .put("node.name", name)
             .put(NodeEnvironment.NODE_ID_SEED_SETTING.getKey(), seed);
 
-        final boolean usingSingleNodeDiscovery = DiscoveryModule.DISCOVERY_TYPE_SETTING.get(finalSettings.build()).equals("single-node");
+        final boolean usingSingleNodeDiscovery = DISCOVERY_TYPE_SETTING.exists(settings) &&
```

Comment:
I see. Super confusing :) can you add a comment that single usage has to be preconfigured? alternatively we can check for ZenDiscovery in the plugin list and the fact that discovery type is not configured/configured to zen. Maybe that's more intuitive as it is a positive test of "is zen discovery active". wdyt?